api/v1alpha1/authorization_types.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// Authorization defines the authorization configuration.
+type Authorization struct {
+	// Rules contains all the authorization rules.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Rules []Rule `json:"rules,omitempty"`
+}
+
+// Rule defines the single authorization rule.
+type Rule struct {
+	// ClientSelectors contains the client selector configuration.
```

Comment:
can we add a little more explanation when multiple clientSelectors are added
* they are ANDd together, only if all selector are valid, then Action is performed